p2p/transport/webrtc/internal/async/condvar.go

```diff
@@ -0,0 +1,46 @@
+package async
+
+import "sync"
+
+// CondVar is an edge-Signaled condition shared by multiple goroutines.  It is
+// analogous in effect to the standard condition variable (sync.Cond) but uses
+// a channel for signaling.
+//
+// To wait on the CondVar, call Wait to obtain a channel. This channel will be
+// closed the next time the CondVar method is executed. A fresh channel is
+// created after each call to CondVar, and is shared among all calls to Wait
+// that occur in the window between signals.
+//
+// A zero CondVar is ready for use but must not be copied after first use.
+type CondVar struct {
+	sync.Mutex
+	ch chan struct{}
+
+	// The signal channel is lazily initialized by the first waiter.
+}
+
+// Signal wakes all pending waiters and resets the CondVar.
+func (cv *CondVar) Signal() {
```

Comment:
This has different semantics then Cond.Signal. It should be called `Broadcast`.

But can we get rid of this whole thing? Do you need the broadcast semantics or
does unblocking one waiter work as well? If one waiter is fine you can just use
a channel as a semaphore (`chan struct{}`)

Reply:
Yes please! I think I've asked for this on one of my previous reviews. Introducing new synchronization primitives isn't something we should do lightly.

Reply:
I agree. This has now been addressed.